### ISSUE TITLE
Mean shift isolated single point issue;

### ISSUE THREAD

**Issue by alexxucui** [mlpack/mlpack#1526]

**Comment by alexxucui** [comment]
@ShangtongZhang Thanks for implementing mean-shift. 

I have a question about one of my use case:
I have three input points of:

(-8.03939 -24.0719) 
(-8.03939 -24.0719) 
(-5.54168 -16.5931)

I set the radius to 0.3. Obviously there should be 2 clusters but the lib gives only 1 cluster. The centroid are at (-8.039389, -24.071899) which makes sense. Even I removed the last point the centroid is still the same. I have tried turn on and off the force convergence but the results is the same. It seems the algorithm will always force one single points to some centroid.

**Comment by rcurtin** [comment]
Hey there @alexxucui, what is `min_points` set to?  Is the third point being labeled as 'noise' instead of a cluster?

**Comment by alexxucui** [comment]
Hi @rcurtin I cannot find `min_points` you mentioned. Do you mean `minFreq` in the `GenSeeds`? And I cannot find 'noise'? I only get assignments and centroids as output.

**Comment by alexxucui** [comment]
@rcurtin Thanks a lot.

**Comment by rcurtin** [comment]
Hi @alexxucui, sorry my response took so long.  It took until today to find a few minutes to look into this one.  I found that the existing mean shfit implementation disallows clusters that hold only a single point, but I don't know any reason for this.  (Perhaps it could be a parameter or something.)

In any case, I opened PR #1536 which should fix the issue.  I can't think of any easy workaround with the existing codebase, so you'll either have to copy the patch into your code manually, wait until the PR is merged, or wait until the next release that incorporates the patch.  (I have been meaning to release 3.0.4 for quite a while now, so I will have to actually find the time to do that in the next few weeks.)

Hope this helps!  Let me know if I overlooked anything.  And thanks for the bug report. :+1:
